src/Utils.hs

```diff
@@ -176,11 +176,13 @@ parens = between (symbol "(") (symbol ")")
 
 -- | 'integer' parses an integer.
 integer :: Parser Integer
-integer = lexeme L.decimal
+unsignedInteger       = lexeme L.decimal
+integer = L.signed sc unsignedInteger
 
 -- | 'float' parses a floating point number.
 float :: Parser Float
-float = realToFrac <$> lexeme L.float -- TODO: parsing without sign?
+unsignedFloat = lexeme L.float
+float = L.signed sc unsignedFloat
```

Comment:
Using `sc` here means that `-     100.0` will parse as `-100.0`. Did you intend for that to be valid? (If so, that's fine)

Reply:
Right, if someone wrote a float, then he wrote a float